### PR TITLE
fix(compile): fix compile when -Werror and -Wunused are enabled

### PR DESCRIPTION
**Pull request by brioja**
Want to contribute? Great! Make sure you've read, understood and considered all
the points below before creating your PR:

- Keep your PR under 200 lines of code and address a single concern.
- Add unit test(s) and documentation (these do not count toward your 200 lines).
- Adhere to the [Google C++ Style Guide](https://google.github.io/styleguide/cppguide.html).
- Run `ninja test` or `catkin_make_isolated --install --use-ninja --pkg cartographer --make-args test` as appropriate.
- Keep rebasing (or merging) of master branch to a minimum. It triggers Travis
  runs for every update which blocks merging of other changes. Our merge bot
  will rebase your branch, reformat your source code and merge as the last step
  in the review process.
- Please replace this template text with the commit message you want for your
  PR. You and/or the reviewer should keep it updated during the course of the
  review using the GitHub edit feature.

### CODE REVIEW

File: cartographer/metrics/family_factory.h [hunk 1]
```diff
@@ -47,7 +47,7 @@ class Family {
 template <typename MetricType>
 class NullFamily : public Family<MetricType> {
  public:
-  MetricType* Add(const std::map<std::string, std::string>& labels) override {
+  MetricType* Add(const std::map<std::string, std::string>&) override {
```
Comment (catskul) [inline cartographer/metrics/family_factory.h]: We should probably preserve the name of the unused parameter at least. Often cstyle comments (bless their hearts) are used for this. Alternately if possible using a pragma to ignore the warning might make sense here.

Comment (catskul) [reply cartographer/metrics/family_factory.h]: This might also work if hideous: `__attribute__((unused))`

Comment (brioja) [reply cartographer/metrics/family_factory.h]:
```suggestion
  MetricType* Add(const std::map<std::string, std::string>& /*labels*/) override {
```
Like this?

**Review by catskul — APPROVED** [review]
@ajakhotia will probably have an opinon and we might need to undo this at some point to help getting our changes upstream, but otherwise works for me.

**Comment by brioja** [comment]
Okay, probably the most cross-platform way would be to use a (void)labels; in the function itself. I could switch it to that?

**Comment by catskul** [comment]
> Okay, probably the most cross-platform way would be to use a (void)labels; in the function itself. I could switch it to that?

Completely unsure. Not sure what the policy is upstream at cartographer/cartographer

**Comment by brioja** [comment]
i guess I'll merge the pull request then. At least it gets it compling for me

**Pull request merged**